site/source/tutorials/1030_routing/index.md

```diff
@@ -0,0 +1,131 @@
+{% endaside %}
+
+An `Outlet` is a higher order component that wraps a widget, and controls whether the widget is rendered based on if the navigated route associated to outlets name. To keep out application organized, outlets are usual stored in a separate directory called `outlets`.
+
+{% instruction 'Add the following code to `WorkerFormOutlet.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/outlets/WorkerFormOutlet.ts' %}
+
+The fist argument for the `Outlet` function determines the widget to display when the configured route is selected. Consider an `outlet` configured for a `path` of `foo`, its wrapped widget will render for a selected route `foo` (described as an `index` match). It will also display for any route that the outlet's `path` partially matches too, for example, `foo/bar` or `foo/bar/baz`.
+
+Normally this is perfectly acceptable, however there are scenarios, where it is necessary to explicitly define a widget for an `index` match. To support this more advanced configuration, first argument also accepts an object that can be used to specify these components explicitly.
+
+```ts
+const ExampleOutlet = Outlet({
+	index: MyIndexComponent,
+	component: MyComponent,
+	error: MyErrorComponent
+}, 'outlet-name');
+```
+
+We use this option for both the `BannerOutlet.ts` and `WorkerContainerOutlet.ts`, as we do not want the widgets to render for routes unless the are an exact match for the selected route.
+
+{% instruction 'Add the following code to `WorkerContainerOutlet.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/outlets/WorkerContainerOutlet.ts' %}
+
+{% instruction 'Add the follow code to `BannerOutlet.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/outlets/BannerOutlet.ts' %}
+
+Now, we can the swap out the widgets for the newly created `outlets`. Notice we haven't needed to changed any of wrapped widgets!
+
+{% instruction 'First, swap the widget imports for the outlet imports in `App.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:8-10 %}
+
+{% instruction 'Then, swap the widget usages for their outlets.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:50-55,59-61 %}
+
+Running the application now, should display the `Banner.ts` by default but also enable routing to the other widgets using the `/#directory` and `/#new-widget` routes.
+
+Next, we will add a side menu with links for the created outlets.
+
+{% section %}
+
+## Adding Links
+
+{% task 'Add a sidebar menu to the application.' %}
+
+In this section we will be using the `Link` component provided by `@dojo/routing`, to create link elements with a `href` for an outlet name.
+
+{% instruction 'Add the `Link` import in `App.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' line:4 %}
+
+{% instruction 'Replace the render function in `App.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:37-55,59-64 %}
```

Comment:
This didn't work for me. After attempting to update structure to match expectations, I got following error in browser's console:

```
Link.ts:54 Uncaught Error: Unable to generate link as injected router could not be found with key 'Symbol()'
    at Link.withRouter (Link.ts:54)
    at WidgetBase.ts:601
    at Array.reduce (<anonymous>)
    at Link.WidgetBase._runBeforeRenders (WidgetBase.ts:600)
    at Link.WidgetBase.__render__ (WidgetBase.ts:414)
    at Projector.WidgetBase._dNodeToVNode (WidgetBase.ts:699)
    at Projector.WidgetBase._dNodeToVNode (WidgetBase.ts:708)
    at Projector.WidgetBase._dNodeToVNode (WidgetBase.ts:708)
    at Projector.WidgetBase._dNodeToVNode (WidgetBase.ts:708)
    at Projector.WidgetBase._dNodeToVNode (WidgetBase.ts:708)
```

This appears to be coming from failing to add the line `projector.setProperties({ registry });` to `main.ts`. This step should be added above when creating routingConfig, ...

Reply:
yup good catch.